beacon_node/beacon_chain/src/data_availability_checker.rs

```diff
@@ -452,6 +296,89 @@ impl<T: EthSpec, S: SlotClock> DataAvailabilityChecker<T, S> {
         self.data_availability_boundary()
             .map_or(false, |da_epoch| block_epoch >= da_epoch)
     }
+
+    /// Persist all in memory components to disk
+    pub fn persist_all(&self) -> Result<(), AvailabilityCheckError> {
+        self.availability_cache.write_all_to_disk()
+    }
+}
+
+pub fn start_availability_cache_maintenance_service<T: BeaconChainTypes>(
+    executor: TaskExecutor,
+    chain: Arc<BeaconChain<T>>,
+) {
+    // this cache only needs to be maintained if deneb is configured
+    if chain.spec.deneb_fork_epoch.is_some() {
+        let overflow_cache = chain.data_availability_checker.availability_cache.clone();
+        executor.spawn(
+            async move { availability_cache_maintenance_service(chain, overflow_cache).await },
+            "otb_verification_service",
+        );
+    }
+}
+
+async fn availability_cache_maintenance_service<T: BeaconChainTypes>(
+    chain: Arc<BeaconChain<T>>,
+    overflow_cache: Arc<OverflowLRUCache<T>>,
+) {
+    let epoch_duration = chain.slot_clock.slot_duration() * T::EthSpec::slots_per_epoch() as u32;
+    loop {
+        match chain
+            .slot_clock
+            .duration_to_next_epoch(T::EthSpec::slots_per_epoch())
+        {
+            Some(duration) => {
+                // this service should run 3/4 of the way through the epoch
+                let additional_delay = (epoch_duration * 3) / 4;
+                tokio::time::sleep(duration + additional_delay).await;
+                debug!(
+                    chain.log,
+                    "Availability cache maintenance service firing";
+                );
+
+                let deneb_fork_epoch = match chain.spec.deneb_fork_epoch {
+                    Some(epoch) => epoch,
+                    None => break, // shutdown service if deneb fork epoch not set
+                };
+                let current_epoch = match chain
+                    .slot_clock
+                    .now()
+                    .map(|slot| slot.epoch(T::EthSpec::slots_per_epoch()))
+                {
+                    Some(epoch) => epoch,
+                    None => continue, // we'll have to try again next time I suppose..
+                };
+
+                if current_epoch < deneb_fork_epoch {
+                    // we are not in deneb yet
+                    continue;
+                }
+
+                let finalized_epoch = chain
+                    .canonical_head
+                    .fork_choice_read_lock()
+                    .finalized_checkpoint()
+                    .epoch;
+                // any data belonging to an epoch before this should be pruned
+                let cutoff_epoch = std::cmp::max(
+                    finalized_epoch + 1,
+                    std::cmp::max(
```

Comment:
Can use `chain.data_availability_checker.data_availability_boundary()` for this

Reply:
The `cutoff_epoch` isn't exactly the same as the `data_availability_boundary`, though I could used the `data_availability_boundary()` function in a subset of the calculation of the `cutoff_epoch`. However, given that it is already necessary to compute the `finalized_epoch`, `current_epoch`, and `deneb_fork_epoch` for previous calculations, it should be faster (and perhaps easier to understand what's happening) to just use them here instead of recomputing them. However, I did adopt using `saturating_sub` here instead of simple subtraction.

What do you think?